.github/scripts/sync_labels.py

```diff
@@ -0,0 +1,114 @@
+    issue_numbers = []
+    for match in matches:
+        issue_numbers.append(match)
+        print(f"Found issue number: {match}")
+    return issue_numbers
+
+
+def get_linked_prs(repo, issue_number, token):
+    query = f"repo:{repo} is:pr is:open linked:issue {issue_number}"
+    url = f"https://api.github.com/search/issues?q={query}"
+    headers = {"Authorization": f"Bearer {token}"}
+    response = requests.get(url, headers=headers)
+    if response.status_code == 200:
+        prs = response.json().get('items', [])
+        return prs
+    else:
+        raise Exception(f"GitHub API returned {response.status_code}: {response.text}")
+
+
+def sync_labels(issue_numbers, number, repo, label_action):
+    print('Sync labels from Issue to linked pull request')
+    github = Github(github_token)
+    for issue_number in issue_numbers:
+        issue_labels = [label.name for label in github.get_repo(repo).get_issue(int(issue_number)).get_labels()]
+        print(issue_labels)
+        if label_action == 'opened':
+            pull_request_labels = [label.name for label in github.get_repo(repo).get_pull(int(number)).get_labels()]
+            for label in issue_labels:
+                if label and label not in pull_request_labels:
+                    github.get_repo(repo).get_pull(int(number)).add_to_labels(label)
+                    print(f"Found and added issue label: {label}")
+        else:
+            pull_request_labels = [label.name for label in github.get_repo(repo).get_pull(int(number)).get_labels() if label.name.startswith('backport/')]
+            issue_backport_label = [label.name for label in github.get_repo(repo).get_issue(int(issue_number)).get_labels() if label.name.startswith('backport/')]
+            filter_pull_request_labels = [label for label in pull_request_labels if label not in issue_backport_label]
+            filter_issue_labels = [label for label in issue_backport_label if label not in pull_request_labels]
+
+            if label_action == 'labeled':
+                for filter_pull_request_label in filter_pull_request_labels:
+                    github.get_repo(repo).get_issue(int(issue_number)).add_to_labels(filter_pull_request_label)
+                for filter_issue_label in filter_issue_labels:
+                    github.get_repo(repo).get_pull(int(number)).add_to_labels(filter_issue_label)
+
+            if label_action == 'unlabeled':
+                if filter_pull_request_labels:
+                    for label in filter_pull_request_labels:
+                        github.get_repo(repo).get_issue(int(number)).remove_from_labels(label)
+                if filter_issue_labels:
+                    for label in filter_issue_labels:
+                        github.get_repo(repo).get_issue(int(issue_number)).remove_from_labels(label)
+
+
+def main():
+    args = parser()
+    if args.is_issue:
+        prs = get_linked_prs(args.repository, args.number, github_token)
```

Comment:
seems it could be only one pr that fix the issue
and you can return in the function directly the pr 'number' instead of for loop below